core/server/api/canary/session.js

```diff
@@ -35,11 +36,23 @@ const session = {
                     auth.session.createSession(req, res, next);
                 });
             });
-        }).catch((err) => {
-            throw new common.errors.UnauthorizedError({
-                message: common.i18n.t('errors.middleware.auth.accessDenied'),
-                err
-            });
+        }).catch(async (err) => {
+            if (!common.errors.utils.isIgnitionError(err)) {
+                throw new common.errors.UnauthorizedError({
+                    message: common.i18n.t('errors.middleware.auth.accessDenied'),
+                    err
+                });
+            }
+
+            if (err.errorType === 'PasswordResetRequiredError') {
+                await api.authentication.generateResetToken({
```

Comment:
@kevinansfield what is your opinion this function call type. It awaits for the response currently but was also thinking to make it "fire-and-forget" style of call. Wdyt?

Reply:
Only issue with "fire-and-forget" is that problems with mail config never surface, just appears like emails never arrived. That's typically only a problem for self-hosters though.